openreview/venue/invitation.py

```diff
@@ -2034,24 +2040,27 @@ def set_custom_stage_invitation(self):
                         'withForum': '${6/content/noteId/value}'
                     }
                 }
-            else:
-                paper_invitation_id = self.venue.get_invitation_id(name=custom_stage.name, prefix='${2/content/replytoSignatures/value}')
-                with_invitation = self.venue.get_invitation_id(name=custom_stage.name, prefix='${6/content/replytoSignatures/value}')
-                reply_to = '${4/content/replyto/value}'
 
         elif custom_stage_reply_type == 'revision':
             if custom_stage_reply_type in ['forum', 'withForum']:
                 raise openreview.OpenReviewException('Custom stage cannot be used for revisions to submissions. Use the Submission Revision Stage instead.')
-            if custom_stage_replyto == 'reviews':
-                invitation_name = self.venue.review_stage.name
-            elif custom_stage_replyto == 'metareviews':
-                invitation_name = self.venue.meta_review_stage.name
-            paper_invitation_id = self.venue.get_invitation_id(name=custom_stage.name, prefix='${2/content/replytoSignatures/value}')
-            with_invitation = self.venue.get_invitation_id(name=invitation_name, number='${6/content/noteNumber/value}')
-            reply_to = None
-            edit_readers = [venue_id, '${2/signatures}']
-            note_readers = None
-            invitees = ['${3/content/replytoSignatures/value}']
+
+        if custom_stage_replyto in ['reviews', 'metareviews']:
+            stage_name = self.venue.review_stage.name if custom_stage_replyto == 'reviews' else self.venue.meta_review_stage.name
+            submission_prefix = venue_id + '/' + self.venue.submission_stage.name + '${2/content/noteNumber/value}/'
+            reply_prefix = stage_name + '${2/content/replyNumber/value}'
+            paper_invitation_id = self.venue.get_invitation_id(name=custom_stage.name, prefix=submission_prefix+reply_prefix)
+            submission_prefix = venue_id + '/' + self.venue.submission_stage.name + '${6/content/noteNumber/value}/'
+            reply_prefix = stage_name + '${6/content/replyNumber/value}'
+            with_invitation = self.venue.get_invitation_id(name=custom_stage.name, prefix=submission_prefix+reply_prefix)
+            reply_to = '${4/content/replyto/value}'
+
+            if custom_stage_reply_type == 'revision':
+                with_invitation = self.venue.get_invitation_id(name=stage_name, number='${6/content/noteNumber/value}')
+                reply_to = None
+                edit_readers = [venue_id, '${2/signatures}']
+                note_readers = None
+                invitees = ['${3/content/replytoSignatures/value}']
```

Comment:
@melisabok This is the only place where we are using `replytoSignatures` and I don't think we can remove it. For a review revision for example, the invitee of the invitation has to be the signatures of the review, unless you see some other way?

Reply:
Do we have review revision invitations in API 2?